File: src/org/kde/kdeconnect/Plugins/MousePadPlugin/KeyInputConnection.java
/*
 * SPDX-FileCopyrightText: 2015 David Edmundson <kde@davidedmundson.co.uk>
 *
 * SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
*/

package org.kde.kdeconnect.Plugins.MousePadPlugin;

import android.view.inputmethod.BaseInputConnection;

class KeyInputConnection extends BaseInputConnection {
    private final KeyListenerView view;

    public KeyInputConnection(KeyListenerView targetView, boolean fullEditor) {
        super(targetView, fullEditor);
        view = targetView;
    }

    @Override
    public boolean commitText(CharSequence text, int newCursorPosition) {
        view.sendChars(text);
        return true;
    }
}

File: src/main/java/org/kde/kdeconnect/plugins/mousepad/KeyInputConnection.kt
/*
 * SPDX-FileCopyrightText: 2015 David Edmundson <kde@davidedmundson.co.uk>
 *
 * SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
*/
package org.kde.kdeconnect.plugins.mousepad

import android.view.inputmethod.BaseInputConnection

internal class KeyInputConnection(private val view: KeyListenerView, fullEditor: Boolean) :
    BaseInputConnection(view, fullEditor) {

    override fun commitText(text: CharSequence, newCursorPosition: Int): Boolean {
        view.sendChars(text)
        return true
    }

}
